src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -341,10 +341,38 @@ HTTPProxy::_handleQuery(shd::QueryRequest&& request) {
     return MakeHTTPResponse<shd::QueryResponse>(sh::Statuses::S501_Not_Implemented("query not implemented"), shd::QueryResponse{});
 }
 
+// Send status with default result
+template <class T>
```

Comment:
there are already helpers to generate responses, called MakeHTTPResponse. Please use those

Reply:
Done